WordPress/gutenberg
# [RNMobile] Enable rendering a block's SVG icon directly from an XML string

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6342

## What?

This PR enables rendering a block's SVG icon directly from an XML string. 

## Why?

When a block's registered, especially if it's registered via metadata, it's possible for the icon to be passed as a string beginning with `"<svg"`. In such cases, the icon would not be displayed, as such a string is not encountered for when registering a block on native.

## How?

`react-native-svg` includes a function, `SvgXml`, that is [useful for parsing XML strings](https://github.com/software-mansion/react-native-svg/blob/main/USAGE.md#use-with-xml-strings). This function is now exported as part of the `@wordpress/primitives` package.

In addition a `SvgXml` mock has been added, following the same pattern as other mocked components.

## Testing Instructions

* See https://github.com/wordpress-mobile/gutenberg-mobile/pull/6342 for an example of when a string is passed as a block's icon.

### Testing Instructions for Keyboard

N/A
